core/capabilities/ccip/configs/solana/contract_reader.go

```diff
@@ -1,11 +1,486 @@
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+	solanacodec "github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
 )
 
-var DestReaderConfig = config.ContractReader{}   //  TODO update the Dest chain reader configuration
-var SourceReaderConfig = config.ContractReader{} // TODO update the Source chain reader configuration
+var ccipOffRampIDL = idl.FetchCCIPOfframpIDL()
+var ccipFeeQuoterIDL = idl.FetchFeeQuoterIDL()
+
+// TODO add events when Querying is finished
+func DestContractReaderConfig() (config.ContractReader, error) {
+	type TimestampedUnixBig struct {
+		Value     *big.Int `json:"value"`
+		Timestamp uint32   `json:"timestamp"`
+	}
+
+	var offRampIDL solanacodec.IDL
+	if err := json.Unmarshal([]byte(ccipOffRampIDL), &offRampIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP OffRamp IDL, error: %w", err)
+	}
+
+	var feeQuoterIDL solanacodec.IDL
+	// TODO ccipFeeQuoterIDL needs to be exported from chainlink-ccip, this is just a placeholder
+	if err := json.Unmarshal([]byte(ccipFeeQuoterIDL), &feeQuoterIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP Fee Quoter IDL, error: %w", err)
+	}
+
+	var routerIDL solanacodec.IDL
+	if err := json.Unmarshal([]byte(ccipRouterIDL), &routerIDL); err != nil {
+		return config.ContractReader{}, fmt.Errorf("unexpected error: invalid CCIP Router IDL, error: %w", err)
+	}
+
+	locationFirst := codec.ElementExtractorLocationFirst
+	return config.ContractReader{
+		AddressShareGroups: [][]string{{consts.ContractNameRouter, consts.ContractNameNonceManager}},
+		Namespaces: map[string]config.ChainContractReader{
+			consts.ContractNameOffRamp: {
+				IDL: offRampIDL,
+				Reads: map[string]config.ReadDefinition{
+					consts.MethodNameGetLatestPriceSequenceNumber: {
+						ChainSpecificName: "GlobalState",
+						ReadType:          config.Account,
+						PDADefinition:     solanacodec.PDATypeDef{Prefix: []byte("state")},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{
+								Fields: map[string]string{"LatestPriceSequenceNumber": "LatestSeqNr"},
+							}},
+					},
+					consts.MethodNameOffRampGetStaticConfig: {
+						ChainSpecificName: "Config",
+						ReadType:          config.Account,
+						PDADefinition: solanacodec.PDATypeDef{
+							Prefix: []byte("config"),
+						},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{
+								Fields: map[string]string{
+									"SvmChainSelector": "ChainSelector",
+								},
+							},
+							&codec.HardCodeModifierConfig{
+								OnChainValues: map[string]any{
+									"GasForCallExactCheck": 0,
+									"RmnRemote":            solana.PublicKey{},
+									// TODO what to do with these two? Do they share address with router?
+									"TokenAdminRegistry": solana.PublicKey{},
```

Comment:
I think this is a [PDA](https://github.com/smartcontractkit/chainlink-ccip/blob/6aca485891a7ccb9e9815e4a19b3d5ec5eb1f2cc/chains/solana/contracts/programs/ccip-router/src/token_context.rs#L33-L40) inside the router